### DISCUSSION TITLE
Regex versioning only compare semver components

### DISCUSSION THREAD

**Discussion by astellingwerf** [renovatebot/renovate#12063] [Request Help] [answered]
The great thing about the regex versioning, is that non-standard versions can still be compared properly. It allowed me to automatically update a docker image from version `11.0.10_9.202107.262569-ol8-jre-rootless` to `11.0.12_7.202109.306039-ol8-jre-rootless`, with a pattern `regex:^(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)_\\d+\\.\\d{4}\\d{2}.\\d*(-(?<compatibility>.*))?$`. The patterns was required because the default docker versioning assumed that everything from the underscore to the first dash was indicating a prerelease version.

Now version `11.0.12_7.202110.315211-ol8-jre-rootless` is released, but Renovate won't suggest it as an update, I guess because the major, minor, patch and compatibility all are the same.
Would it not make sense to allow for the additional comparison of an arbitrary number of version/build numbers?

**Comment by viceice** [comment, marked as answer]
You can already use a `build` capture group as an additional number.

@HonkingGoose can you propose a PR to add it to the docs here:

https://github.com/renovatebot/renovate/tree/main/lib/versioning/regex

I think I forgot it when implemented the feature

**Reply by HonkingGoose** [reply]
> I think I forgot it when implemented the feature

@viceice I think this PR implemented the feature. 😉  [^pr]

[^pr]: https://github.com/renovatebot/renovate/pull/11115

**Reply by HonkingGoose** [reply]
See PR #12065 for the documentation fix. 😉
